docs/log-levels.md

```diff
@@ -202,7 +203,7 @@ These are only general guidelines, and there always will be exceptions to these
 
 Here are a few examples of situations when logging a message on a relatively high level might still be tolerable for a library.
 
-It's permissible for a library to log at `critical` level right before a _hard crash_, as a last resort of informing the log collection systems or end-user about additional information detailing the reason for the crash. This should be _in addition to_ the message from a `fatalError` and can lead to an improved diagnosis/debugging experience for end users.
+It's permissible for a library to log at `critical` level right before a _hard crash_ or assertion failure, as a last resort of informing the log collection systems or end-user about additional information detailing the reason for the crash. This should be _in addition to_ the message from a `fatalError` and can lead to an improved diagnosis/debugging experience for end users.
```

Comment:
Adding "assertion failure" here as a critical failure and system exit may happen under some circumstances where there has been a corruption of a data structure or file, and it is safer for the system to exit rather than keep running.  This isn't a "hard crash" technically, and the assertion violation itself may not be couched in terms that the user understands, so the critical error message may be a "high level overview" that explains that why the system is exiting.

Reply:
I see your point but I think this is mixing up terminology of the two (java/scala and swift) worlds. I would not want to say assertion error because assertions are a specific thing in swift and they are not enabled in release builds. If anything this would have to be "precondition failure".

I've been pretty consistently pushing for "hard crash" meaning "process exit" in many crash handling discussions in the community as we'd like to get panics or "soft crash" (say, "crash the actor" or "thread", with a panic or similar), those would be soft crashes.

I'll keep this bit as is I think but thank you for the feedback!

Reply:
```suggestion
It's permissible for a library to log at `critical` level right before a _hard_ crash of the process, as a last resort of informing the log collection systems or end-user about additional information detailing the reason for the crash. This should be _in addition to_ the message from a `fatalError` and can lead to an improved diagnosis/debugging experience for end users.
```